text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/Placeholder.java

```diff
@@ -21,74 +21,74 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-package net.kyori.adventure.text.minimessage.placeholder;
+package net.kyori.adventure.text.minimessage.tag;
 
 import java.util.Objects;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.ComponentLike;
-import org.jetbrains.annotations.ApiStatus;
+import net.kyori.adventure.text.minimessage.tag.resolver.TagResolver;
 import org.jetbrains.annotations.NotNull;
 
+import static java.util.Objects.requireNonNull;
+
 /**
- * A replacement with an associated key.
+ * Tag resolvers producing tags that insert fixed values.
+ *
+ * <p>These are effectively placeholders.</p>
  *
- * @param <T> the type of the replacement
  * @since 4.10.0
  */
-@ApiStatus.NonExtendable
-public interface Placeholder<T> extends Replacement<T> {
+public final class Placeholder {
+  private Placeholder() {
+  }
+
   /**
-   * Creates a placeholder that inserts a MiniMessage string. The inserted string will impact
-   * the rest of the parse process.
+   * Creates a placeholder that inserts a MiniMessage string.
+   *
+   * <p>The inserted string will impact the rest of the parse process.</p>
    *
    * @param key the key
    * @param value the replacement
    * @return the placeholder
    * @since 4.10.0
    */
-  static @NotNull Placeholder<String> miniMessage(final @NotNull String key, final @NotNull String value) {
-    PlaceholderImpl.checkKey(key);
-    return new PlaceholderImpl<>(key, Objects.requireNonNull(value, "value"));
+  public static TagResolver.@NotNull Single parsed(final @NotNull String key, final @NotNull String value) {
+    return TagResolver.resolver(key, Tag.preProcessParsed(value));
   }
 
   /**
-   * Creates a placeholder that inserts a raw string, ignoring any MiniMessage tags present.
+   * Creates a placeholder that inserts a literal string, without attempting to parse any contained tags.
    *
    * @param key the key
    * @param value the replacement
    * @return the placeholder
    * @since 4.10.0
    */
-  static @NotNull Placeholder<Component> raw(final @NotNull String key, final @NotNull String value) {
-    return component(key, Component.text(value));
+  public static TagResolver.@NotNull Single unparsed(final @NotNull String key, final @NotNull String value) {
+    requireNonNull(value, "value");
+    return Placeholder.component(key, Component.text(value));
   }
 
   /**
    * Creates a replacement that inserts a component.
    *
+   * <p>This replacement is auto-closing, so its style will not influence the style of following components.</p>
```

Comment:
Might it be worth adding a version that allows styles to bleed - would be useful for prefixes with role colours for example.

Reply:
Wouldn't that require significant restructuring? Pretty sure this comment is just clarifying what was the case before, right?

Reply:
This clarifies the behavior before, which is determined by `Inserting#allowsChildren` -- in this case it's `false`, but when creating a `Tag` with `Tag.inserting` the tag *does* allow children. I'm just not sure whether to have that be a boolean parameter or two separate methods